docs/guide-disappearance.md

```diff
@@ -3,93 +3,114 @@ id: guide-disappearance
 title: Appearance and Disappearance
 ---
 
-Sometimes you need to test that an element is present and then disappears or
-vice versa.
+Sometimes you need to test that an element is present and then disappears or the
+other way around.
 
 ## Waiting for appearance
 
-If you need to wait for an element to appear, the [async wait
-utilities][async-api] allow you to wait for an assertion to be satisfied before
-proceeding. The wait utilities retry until the query passes or times out.
+If you need to wait for an element to appear, [`findBy`][find-by] and the [async
+wait utility][async-api] `waitFor` allow you to wait for an assertion to be
+satisfied before proceeding. The wait utilities retry until the query passes or
+times out.
 
-```jsx
+Wait for appearance and return the element using `findBy`:
+
+```javascript
+test('movie title appears', async () => {
+  // element is initially not present...
+
+  const movieTitle = await screen.findByText('the lion king')
+  expect(movieTitle).toBeInTheDocument()
+})
+```
+
+Or wait for appearance using `waitFor` and `getBy`:
+
+```javascript
 test('movie title appears', async () => {
   // element is initially not present...
 
-  // wait for appearance
   await waitFor(() => {
-    expect(getByText('the lion king')).toBeInTheDocument()
+    expect(screen.getByText('the lion king')).toBeInTheDocument()
   })
-
-  // wait for appearance and return the element
-  const movie = await findByText('the lion king')
 })
 ```
 
+These do more or less the same thing (`findBy` uses `waitFor` under the hood),
+but the `findBy` version results in simpler code and a better error message.
```

Comment:
Okay, this is interesting.

I'm not sure I agree on the statement (I believe using `waitFor` is more explicit, and helps the test conveying meaning) but that's entirely personal. The question here is whether if docs should showcase two ways of doing the same thing 🤔 

If we agree on `findBy` being simpler and packed with a better error message, then I'd say this is the way docs should suggest 👍 

`waitFor` is useful for other stuff (such as waiting for XHR requests to happen, or other). It might appear somewhere else in docs.

Reply:
I think this is the last bit that needs some attention, then we can merge!